assets/src/apps/delivery/store/features/groups/actions/deck.ts

```diff
@@ -496,6 +496,15 @@ export const loadActivities = createAsyncThunk(
         hasMoreHints: result.hasMoreHints || true,
         groupId: null,
       };
+      //To improve the performance, when a lesson is opened in authoring, we generate a list of variables that contains expression and needs evaluation
+      // we stored them in conditionsNeedEvaluation in activity.content.custom.conditionsNeedEvaluation. When this function is called
+      // we only process variables that is present in conditionsNeedEvaluation array and ignore others.
```

Comment:
what is going to be done about things that haven't had this in the authoring tool? is there a migration needed?

Reply:
I already mentioned that the lesson needs to be open once in the authoring tool in order this to work correctly. Apart from the Janus-text, other areas like Init State, Adaptive Rules, ConfigData is covered here.